lib/logwriter.c

```diff
@@ -1100,7 +1100,7 @@ log_writer_write_message(LogWriter *self, LogMessage *msg, LogPathOptions *path_
   if (!(msg->flags & LF_INTERNAL))
     {
       msg_debug("Outgoing message",
-                evt_tag_printf("message", "%.*s", (gint) self->line_buffer->len, self->line_buffer->str));
+                evt_tag_printf("message", "%s", self->line_buffer->str));
     }
 
```

Comment:
if the message is NUL terminated (is it?), then why can't we use evt_tag_str() here?

BTW: I think it is not guaranteed to be NUL terminated, although possibly this is ok, as we are not perfectly NUL safe.

But with $(binary 0x00) you can embed a NUL character in the output stream.

Reply:
I think it is perfectly fine to truncate in an internal message, especially if it is about our outgoing stream.

So, it is fine as it is, I just wanted to mention that evt_tag_str() should be ok as well. That doesn't have the truncation as a feature though.

Reply:
We use evt_tag_printf with a `%.*s` format string in several places, just to truncate the output.
I think the format string is quite a big overhead to just use it as a "truncate" feature.

I propose a new function:
evt_tag_strn(const char* s, int limit, int length = -1)

If ok, I will create a new PR, and replace the evt_tag_printf in the code. (Including this one.)